tests/storages_tests/test_heartbeat.py

```diff
@@ -49,7 +49,7 @@ def test_fail_stale_trials_with_optimize(storage_mode: str) -> None:
         with patch("optuna.study._optimize.Thread", _TestableThread):
             study1.optimize(lambda _: 1.0, n_trials=1)
 
-        assert study1.trials[0].state is TrialState.FAIL
+        assert study1.trials[0].state is TrialState.FAIL  # type: ignore [comparison-overlap]
```

Comment:
[Notes]  `mypy` inferred that `study1.trials[0].state` was `Literal[TrialState.RUNNING]` due to the assertion at L45.